lib/cli/interactive-setup/utils.js

```diff
@@ -13,4 +18,33 @@ module.exports = {
       })
     )[name];
   },
+  isUserLoggedIn: () => {
+    if (process.env.SERVERLESS_ACCESS_KEY) {
+      return true;
+    }
+
+    if (configUtils.getLoggedInUser()) {
+      return true;
+    }
+
+    return false;
+  },
+  doesServiceInstanceHaveLinkedProvider: async ({ configuration, options }) => {
+    const region = resolveRegion({ configuration, options });
+    const stage = resolveStage({ configuration, options });
+    let result;
+    try {
+      result = await resolveProviderCredentials({ configuration, region, stage });
+    } catch (err) {
+      if (err.statusCode && err.statusCode >= 500) {
+        throw new ServerlessError(
+          'Dashboard service is currently unavailable, please try again later',
+          'DASHBOARD_UNAVAILABLE'
+        );
+      }
+      throw err;
+    }
+
+    return Boolean(result);
+  },
```

Comment:
Those utils feel to me as `dashboard-plugin` utils, they're not specific to interactive CLI

Reply:
Do you suggest to move them to `dashboard-plugin` then?

Reply:
Yes, I think it's a better fit, and I can imagine they will then become useful in other logic points

Reply:
As discussed previously, I've switched to use `isAuthenticated` from the plugin but kept the other util here as we would still have to develop a similar wrapper for that functionality on Framework side.